Modules/xASL_module_ASL.m

```diff
@@ -132,23 +144,147 @@
+    warning('Missing field x.Q.Initial_PLD, this is needed for ASL quantification');
+elseif isempty(x.Q.Initial_PLD) || ~isnumeric(x.Q.Initial_PLD)
+    warning('Illegal field x.Q.Initial_PLD, this should not be empty and should contain numerical values');
 else
-    x.modules.asl.bMultiPLD = false;
+    x.Q.uniqueInitial_PLD = unique(x.Q.Initial_PLD);
+    x.Q.NumberPLDs = length(x.Q.uniqueInitial_PLD);
+    
+    if sum(x.Q.Initial_PLD<=0)>0
+        warning('x.Q.Initial_PLD should be positive');
+    end
+
+    % Handle different number of PLDs
+    if x.Q.NumberPLDs==1
+        fprintf('%s\n', 'Single-PLD ASL detected');
+    elseif x.Q.NumberPLDs>1
+        fprintf('%s\n', 'Multi PLDs detected, processing this as multi-PLD ASL');
+        fprintf('%s\n', 'Note that this feature is still under development');
+        x.modules.asl.bMultiPLD = true;
+    end
+
+    fprintf('%s\n', 'Detected the following unique PLDs (ms):')
+    for iPLD = 1:x.Q.NumberPLDs-1
+        fprintf('%d, ',x.Q.uniqueInitial_PLD(iPLD));
+    end
+    fprintf('%d\n',x.Q.uniqueInitial_PLD(end));
+end
+
+% Deal with too short vectors (e.g., repetitions in the case of single-PLD)
+nPLD = length(x.Q.Initial_PLD);
+factorPLD = nVolumes/nPLD;
+x.Q.InitialPLD = repmat(x.Q.Initial_PLD(:), [factorPLD 1]);
+
+% Number of initial PLDs should be equal to number of ASL volumes
+if nPLD~=nVolumes
+    warning(['Number of PLDs (n=' xASL_num2str(nPLD) ') should be equal to number of ASL volumes (n=' xASL_num2str(nVolumes) ')']);
+end
+
+%% D5. Labeling Duration parsing
+
+if ~isfield(x.Q,'LabelingDuration')
+    warning('Missing field x.Q.LabelingDuration, this is needed for ASL quantification');
+elseif isempty(x.Q.LabelingDuration) || ~isnumeric(x.Q.LabelingDuration)
+    warning('Illegal field x.Q.LabelingDuration, this should not be empty and should contain numerical values');
+else
+    x.Q.uniqueLabelingDuration = unique(x.Q.LabelingDuration);
+    x.Q.NumberLDs = length(x.Q.uniqueLabelingDuration);
+    
+    if sum(x.Q.LabelingDuration<=0)>0
+        warning('x.Q.LabelingDuration should be positive');
+    end
+
+    % Handle different number of Labeling Durations
+    if x.Q.NumberLDs==1
+        fprintf('%s\n', 'Single-labeling duration ASL detected');
+    elseif x.Q.NumberLDs>1 && x.modules.asl.bMultiPLD
+        fprintf('%s\n', 'Multi labeling durations detected, taking this into account for multi-PLD ASL processing');
+    elseif x.Q.NumberLD>1 && ~x.modules.asl.bMultiPLD
+        warning('Multi labeling durations detected but not multiple PLDs, is this correct?');
+    end
+
+    fprintf('%s\n', 'Detected the following unique Labeling Durations (ms):')
+    for iLD = 1:x.Q.NumberLDs-1
+        fprintf('%d, ',x.Q.uniqueLabelingDuration(iLD));
+    end
+    fprintf('%d\n',x.Q.uniqueLabelingDuration(end));
 end
 
-%% D4. TimeEncoded parsing
+% Deal with too short vectors (e.g., repetitions in the case of single-PLD)
+nLD = length(x.Q.LabelingDuration);
```

Comment:
Please do the same fix as above.

Reply:
Done in 0bdf54eb502121e43a25db8167411a91e183d051